nifi-extension-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/JWTBearerOAuth2AccessTokenProvider.java

```diff
@@ -0,0 +1,677 @@
+        String algorithmName = validationContext.getProperty(SIGNING_ALGORITHM).getValue();
+        PrivateKey privateKey = keyService.getPrivateKey();
+        JWSAlgorithm algorithm = JWSAlgorithm.parse(algorithmName);
+
+        if (validationContext.getProperty(HEADER_X5T).asBoolean() && !(privateKey instanceof RSAPrivateKey)) {
+            validationResults.add(new ValidationResult.Builder()
+                    .subject(HEADER_X5T.getDisplayName())
+                    .valid(false)
+                    .explanation("The private key must be an RSA key to set the x5t header.")
+                    .build());
+        }
+
+        if (privateKey instanceof RSAPrivateKey) {
+            if (algorithmName.startsWith("RS") || algorithmName.startsWith("PS")) {
+                return validationResults;
+            } else {
+                validationResults.add(new ValidationResult.Builder()
+                        .subject(SIGNING_ALGORITHM.getDisplayName())
+                        .valid(false)
+                        .explanation("The private key is of RSA type, the signing algorithm must be either RS256 or PS256.")
+                        .build());
+            }
+        } else if (privateKey instanceof ECPrivateKey) {
+            if (algorithm.equals(JWSAlgorithm.ES256) || algorithm.equals(JWSAlgorithm.ES384) || algorithm.equals(JWSAlgorithm.ES512)) {
+                return validationResults;
+            } else {
+                validationResults.add(new ValidationResult.Builder()
+                        .subject(SIGNING_ALGORITHM.getDisplayName())
+                        .valid(false)
+                        .explanation("The private key is of EC type, the signing algorithm must be either ES256, ES384 or ES512.")
+                        .build());
+            }
+        } else if (algorithm.equals(JWSAlgorithm.Ed25519)) {
+            return validationResults;
+        } else {
+            validationResults.add(new ValidationResult.Builder()
+                    .subject(SIGNING_ALGORITHM.getDisplayName())
+                    .valid(false)
+                    .explanation(String.format("The provided key (algorithm = %s) is not supported for signing algorithm %s", privateKey.getAlgorithm(), algorithmName))
+                    .build());
+        }
+
+        return validationResults;
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(ConfigurationContext context, ComponentLog verificationLogger, Map<String, String> variables) {
+        getProperties(context);
+        initJWTSigner();
+        ConfigVerificationResult.Builder builder = new ConfigVerificationResult.Builder().verificationStepName("Acquire token");
+        try {
+            getAccessDetails();
+            builder.outcome(ConfigVerificationResult.Outcome.SUCCESSFUL);
+        } catch (Exception ex) {
```

Comment:
Minor convention recommendation:
```suggestion
        } catch (final Exception e) {
```